test/e2e/allocate.go

```diff
@@ -0,0 +1,289 @@
+	})
+
+	It("schedule job when resource is NOT enough", func() {
+		ctx := initTestContext(options{
+			nodesNumLimit: 2,
+			nodesResourceLimit: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("2000m"),
+				corev1.ResourceMemory: resource.MustParse("2048Mi")},
+		})
+		defer cleanupTestContext(ctx)
+
+		slot1 := corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("500m"),
+			corev1.ResourceMemory: resource.MustParse("512Mi")}
+		slot2 := corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("1000m"),
+			corev1.ResourceMemory: resource.MustParse("1024Mi")}
+		slot3 := corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("2000m"),
+			corev1.ResourceMemory: resource.MustParse("2048Mi")}
+
+		job := &jobSpec{
+			tasks: []taskSpec{
+				{
+					img: defaultNginxImage,
+					req: slot2,
+					min: 1,
+					rep: 1,
+				},
+			},
+		}
+
+		job.name = "middle"
+		midReqJob := createJob(ctx, job)
+		err := waitJobReady(ctx, midReqJob)
+		Expect(err).NotTo(HaveOccurred())
+
+		job.name = "low"
+		job.tasks[0].req = slot1
+		lowReqJob := createJob(ctx, job)
+		err = waitJobReady(ctx, lowReqJob)
+		Expect(err).NotTo(HaveOccurred())
+
+		job.name = "high"
+		job.tasks[0].req = slot3
+		highReqJob := createJob(ctx, job)
+		err = waitJobPending(ctx, highReqJob)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = compareNodename(ctx, lowReqJob, midReqJob)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("allocate don't work when podgroup is pending", func() {
+		ctx := initTestContext(options{
+			nodesNumLimit: 2,
+			nodesResourceLimit: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("2000m"),
+				corev1.ResourceMemory: resource.MustParse("2048Mi")},
+		})
+		defer cleanupTestContext(ctx)
+
+		pgName := "pending-pg"
+		pg := &schedulingv1beta1.PodGroup{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: ctx.namespace,
+				Name:      pgName,
+			},
+			Spec: schedulingv1beta1.PodGroupSpec{
+				MinMember:    1,
+				MinResources: &thirtyCPU,
+			},
+			Status: schedulingv1beta1.PodGroupStatus{
+				Phase: schedulingv1beta1.PodGroupPending,
+			},
+		}
+
+		_, err := ctx.vcclient.SchedulingV1beta1().PodGroups(ctx.namespace).Create(context.TODO(), pg, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		slot1 := corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("500m"),
+			corev1.ResourceMemory: resource.MustParse("512Mi")}
+
+		job := &jobSpec{
+			tasks: []taskSpec{
+				{
+					img: defaultNginxImage,
+					req: slot1,
+					min: 1,
+					rep: 1,
+				},
+			},
+		}
+
+		job.name = "exist"
+		existJob := createJob(ctx, job)
+		err = waitJobReady(ctx, existJob)
+		Expect(err).NotTo(HaveOccurred())
+
+		job.name = "allocate"
+		allocateJob := createJobWithCondition(ctx, job, pgName, nil)
+		err = waitJobPending(ctx, allocateJob)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("allocate don't work when queue NOT exist", func() {
```

Comment:
I donot think this case will test anything on scheduling, the job creation is denied by admission.
So i would suggest removing it

Reply:
I agree with you. I'll remove it.